services/dataplatform/cluster.go

```diff
@@ -0,0 +1,192 @@
+package dataplatform
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	dataplatform "github.com/ionos-cloud/sdk-go-dataplatform"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+	"log"
+	"strings"
+)
+
+var clusterResourceName = "Dataplatform Cluster"
+
+func (c *Client) IsClusterReady(ctx context.Context, d *schema.ResourceData) (bool, error) {
+	cluster, _, err := c.GetClusterById(ctx, d.Id())
+	if err != nil {
+		return false, err
+	}
+	if cluster.Metadata == nil || cluster.Metadata.State == nil {
+		return false, fmt.Errorf("expected metadata, got empty for id %s", d.Id())
+	}
+	log.Printf("[DEBUG] dataplatform cluster state %s", *cluster.Metadata.State)
+	return strings.EqualFold(*cluster.Metadata.State, utils.Available), nil
+}
+
+func (c *Client) GetClusterById(ctx context.Context, id string) (dataplatform.ClusterResponseData, *dataplatform.APIResponse, error) {
+	cluster, apiResponse, err := c.sdkClient.DataPlatformClusterApi.GetCluster(ctx, id).Execute()
+	apiResponse.LogInfo()
+	return cluster, apiResponse, err
+}
+
+// IsClusterDeleted - returns apiResponse to check if resource still exists. To be used with WaitForResourceToBeDeleted
```

Comment:
It's tricky, code should be commented only when it's not clear what it does. I added the comment here because I want to show how the function is used with WaitForResourceToBeDeleted. Even here, the comment is no longer in sync with what the function does.

Reply:
@cristiGuranIonos `It's tricky, code should be commented only when it's not clear what it does` I agree, sorry, I didn't actually read the code I only looked if we have function docs or not.

`Even here, the comment is no longer in sync with what the function does.` -- Then maybe we should sync it again by updating the comment?

Reply:
yes, already done